File: sol/1498/1498.cpp
struct Solution {
    int numSubseq(vector<int>& nums, int target) {
        constexpr int M = 1e9 + 7;
        int n = nums.size(), res = 0;
        vector<int> power(n + 1);
        power[0] = 1;
        for (int i = 1; i <= n; ++i){
            power[i] = (power[i - 1] * 2) % M;
        }
        ranges::sort(nums);
        int r = n - 1;
        for (int l = 0; l < n; ++l){
            while (r >= l && nums[l] + nums[r] > target){
                --r;
            }
            if (r < l){
                break;
            }
            res = (res + power[r - l]) % M;
        }
        return res;
    }
};

File: sol/1498/1498.py
class Solution:
    def numSubseq(self, nums: List[int], target: int) -> int:
        M = 10 ** 9 + 7
        n = len(nums)
        nums.sort()
        res = 0
        r = n - 1
        for l in range(n):
            while r >= l and nums[l] + nums[r] > target:
                r -= 1
            if r < l:
                break
            res = (res + pow(2, r - l, M)) % M
        return res
